src/client/src/orangehrmCorporateDirectoryPlugin/pages/CorporateDirectory.vue

```diff
@@ -0,0 +1,255 @@
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <div class="orangehrm-background-container">
+    <oxd-table-filter :filter-title="$t('Directory')">
+      <oxd-form>
+        <oxd-form-row>
+          <oxd-grid :cols="3" class="orangehrm-full-width-grid">
+            <oxd-grid-item>
+              <employee-autocomplete />
+            </oxd-grid-item>
+            <oxd-grid-item>
+              <jobtitle-dropdown />
+            </oxd-grid-item>
+            <oxd-grid-item>
+              <oxd-input-field
+                :clear="false"
+                :label="$t('general.location')"
+                :options="countries"
+                type="select"
+              />
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+
+        <oxd-divider />
+
+        <oxd-form-actions>
+          <oxd-button
+            :label="$t('general.reset')"
+            display-type="ghost"
+            type="reset"
+          />
+          <oxd-button
+            :label="$t('general.search')"
+            class="orangehrm-left-space"
+            display-type="secondary"
+            type="submit"
+          />
+        </oxd-form-actions>
+      </oxd-form>
+    </oxd-table-filter>
+    <br />
+    <div class="orangehrm-corporate-directory">
+      <div class="orangehrm-paper-container">
+        <table-header
+          :loading="isLoading"
+          :show-divider="false"
+          :total="total"
+        ></table-header>
+        <oxd-grid :cols="colSize" class="orangehrm-container">
+          <oxd-grid-item v-for="employee in employees" :key="employee">
+            <summary-card
+              :id="employee.id"
+              :employee-designation="employee.employeeJobTitle"
+              :employee-location="employee.employeeLocation"
+              :employee-name="employee.employeeName"
+              :employee-sub-unit="employee.employeeSubUnit"
+              :show-employee-details="isMobile ? toggle : false"
+              @click="
+                isMobile === false ? showEmployeeDetailsFn(employee.id) : '';
+                toggle = !toggle;
+              "
+            ></summary-card>
+          </oxd-grid-item>
+        </oxd-grid>
+        <div class="orangehrm-bottom-container"></div>
+      </div>
+      <div
+        v-if="showDetails && isMobile === false"
+        class="orangehrm-paper-container orangehrm-corporate-directory-sidebar"
+      >
+        <oxd-grid-item>
+          <summary-card-details
```

Comment:
What if rather than querying data here, you pass the employee id to summary card, and when it's loaded it will load with data from inside it?

Reply:
done